[GLib] Add a websetting gating WebRTC support

philn opened the pull request:
#### feef4553836dd7f7126823f41a21d60c0f9481b3
<pre>
[GLib] Add a websetting gating WebRTC support
<a href="https://bugs.webkit.org/show_bug.cgi?id=242103">https://bugs.webkit.org/show_bug.cgi?id=242103</a>

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

WebRTC support in WPE and GTK ports is now opt-in through the enable-webrtc web-setting.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webKitSettingsConstructed):
(webKitSettingsSetProperty):
(webKitSettingsGetProperty):
(webkit_settings_class_init):
(webkit_settings_set_enable_media_stream):
(webkit_settings_get_enable_webrtc):
(webkit_settings_set_enable_webrtc):
* Source/WebKit/UIProcess/API/gtk/WebKitSettings.h:
* Source/WebKit/UIProcess/API/wpe/WebKitSettings.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings):

Canonical link: <a href="https://commits.webkit.org/251951@main">https://commits.webkit.org/251951@main</a>
</pre>

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -1559,6 +1563,23 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
+     * WebKitSettings:enable-webrtc:
+     *
+     * Enable or disable support for WebRTC on pages. Enabling this setting implies MediaStream will
+     * be enabled as well.
```
I would split the second phrase after the period to a new paragraph, so `gi-docgen` will pick only “Enable WebRTC support for loaded pages” as the property summary, otherwise the summary will be too long. We can also link to the `enable-media-stream` property:

```suggestion
     * Enable WebRTC support for loaded pages.
     *
     * Enabling this setting implies that [property@Settings:enable-media-stream]
     * will be enabled as well.
```

philn replied:
I didn't know we're supposed to use gi-docgen syntax now. Isn't that compatible with gtkdoc syntax?

aperezdc replied:
Yes, since [249161@main](https://commits.webkit.org/249161@main). There's a good amount of compatibility with gtk-doc syntax indeed, but it would be good to already use documentation style more suitable for gi-docgen in new code.

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -3263,10 +3284,51 @@ void webkit_settings_set_enable_media_stream(WebKitSettings* settings, gboolean
+ * webkit_settings_get_enable_webrtc:
+ * @settings: a #WebKitSettings
+ *
+ * Get the #WebKitSettings:enable-webrtc property.
```
Use `[property@Settings:enable-webrtc]` for [linking things with gi-docgen](https://gnome.pages.gitlab.gnome.org/gi-docgen/linking.html).

philn replied:
I don't see this syntax used elsewhere. I'm confused :)

aperezdc replied:
It's because we haven't been converting everything in bulk—so far we did a rather minimal conversion; I have plans to work on this kind of fixups in the next months 😄

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -3263,10 +3284,51 @@ void webkit_settings_set_enable_media_stream(WebKitSettings* settings, gboolean
+ * @enabled: Value to be set
+ *
+ * Set the #WebKitSettings:enable-webrtc property. Setting this property to %TRUE implies the
+ * media-stream web-setting will also be enabled.
```
Same comment here: please keep the first phrase brief, which is used as summary; and use `[property@...]` for linking.

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -1559,6 +1563,23 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
+        "enable-webrtc",
+        _("Enable WebRTC"),
+        _("Whether WebRTC content should be handled"),
+        FALSE,
```
Should this be `TRUE` by default? Or does it seem better to keep it as `FALSE` for now and change it if/when we feel that the WebRTC implementation is ready for general consumption?

philn replied:
For now I think FALSE is better, because the GstWebRTC has too many unimplemented features to be usable. I would re-assess this when we are closer to the 2.38 release.

aperezdc reviewed (changes requested):
I like the idea and the patch is mostly fine—I only wrote a couple of comments about the documentation.

aperezdc reviewed (approved):
Thanks @philn! As you are a reviewer yourself, that can count as the second r+ for new API; but feel free to have someone else take a look as well if you would be more comfortable that way 😃

philn committed:
[GLib] Add a websetting gating WebRTC support
https://bugs.webkit.org/show_bug.cgi?id=242103

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

WebRTC support in WPE and GTK ports is now opt-in through the enable-webrtc web-setting.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webKitSettingsConstructed):
(webKitSettingsSetProperty):
(webKitSettingsGetProperty):
(webkit_settings_class_init):
(webkit_settings_set_enable_media_stream):
(webkit_settings_get_enable_webrtc):
(webkit_settings_set_enable_webrtc):
* Source/WebKit/UIProcess/API/gtk/WebKitSettings.h:
* Source/WebKit/UIProcess/API/wpe/WebKitSettings.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings):

Canonical link: https://commits.webkit.org/251951@main
